nmrglue/fileio/sparky.py

```diff
@@ -500,6 +501,40 @@ def write_3D(filename, dic, data, overwrite=False):
     f.close()
     return
 
+def read_4D(filename):
+    """
+    Read a 4D Sparky file. See :py:func:`read` for documentation.
+    """
+    seek_pos = os.stat(filename).st_size
+    f = open(filename, 'rb')
```

Comment:
A context manager should be used here in the same manner as the read_3d function.  e.g.
``` python
    with open(filename, 'rb') as f:

        # read the file header
        dic = fileheader2dic(get_fileheader(f))
````